common-npm-packages/artifacts-common/serviceConnectionUtils.ts

```diff
@@ -87,12 +87,28 @@ export class PrivateKeyServiceConnection extends ServiceConnection {
     }
 }
 
+export class EntraServiceConnection extends ServiceConnection {
+    constructor(
+        public packageSource: IExternalPackageSource,
+        public entraConnectionName: string,
+        public token: string,
```

Comment:
The token will not actually be in the the service connection. The Entra wif service connection only holds the MI /SP informatation. The method here: https://github.com/microsoft/azure-pipelines-tasks-common-packages/blob/3d9a0fbbb26f932bf0498f6c06113592bbea1878/common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts#L12 actually uses the MI/SP information to do the WIF token handshake to get the token. So for the entra types, you will need to call the above function to actually get the token